src/Metrics/DefaultMetricsBucket.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace Unleash\Client\Metrics;
+
+use DateTimeInterface;
+use JetBrains\PhpStorm\ArrayShape;
+use JsonSerializable;
+use LogicException;
+
+/**
+ * @internal
+ */
+final class DefaultMetricsBucket implements MetricsBucket, JsonSerializable
+{
+    /**
+     * @var array<MetricsBucketToggle>
+     */
+    private array $toggles = [];
+
+    public function __construct(
+        private readonly DateTimeInterface $startDate,
+        private ?DateTimeInterface $endDate = null,
+    ) {
+    }
+
+    public function addToggle(MetricsBucketToggle $toggle): self
+    {
+        $this->toggles[] = $toggle;
+
+        return $this;
+    }
+
+    public function getStartDate(): DateTimeInterface
+    {
+        return $this->startDate;
+    }
+
+    public function setEndDate(?DateTimeInterface $endDate): DefaultMetricsBucket
```

Comment:
Can you return `self`? I noticed that it's a mistake carried over from my code, but anyway it would be better to make it consistent.

```suggestion
    public function setEndDate(?DateTimeInterface $endDate): self
```